packages/flutter_tools/lib/src/base/user_messages.dart

```diff
@@ -220,6 +220,10 @@ class UserMessages {
       "matching '$deviceId'";
   String get flutterNoDevicesFound => 'No devices found';
   String get flutterNoSupportedDevices => 'No supported devices connected.';
+  String flutterMissPlatformProject(String flutterProject, List<String> unsupportedDevicesType) =>
+      'Your project $flutterProject is missing some platform projects.'
```

Comment:
I still think "Your project is missing some platform projects." is too strong.  The vast majority of current Flutter users only intend to support Android or iOS.  We don't want to imply there's an error in their project when there isn't one, and they just need to plug in a missing device.

How about:
```
String flutterGenerateAdditionalPlatforms(List<String> unsupportedDevicesType) =>
'If you would like your app to run on ${unsupportedDevicesType.join(' or ')}, consider running `flutter create .` to generate projects for these platforms.';
```
(`FlutterProject.current()` can be extremely expensive to call, we should avoid if possible.

Reply:
ok done.